src/ragapp/backend/agents/orchestrator.py

```diff
@@ -19,7 +19,7 @@ def get_tool(tool_name: str, config: dict, query_engine=None):
         description = f"{config.description or ''}\nThis is a preferred tool to use"
         return QueryEngineTool(
             query_engine=query_engine,
-            metadata=ToolMetadata(name=config.name, description=description),
+            metadata=ToolMetadata(name=tool_name, description=description),
```

Comment:
We're mixing the usage of tool name and tool id. I'll have a refactor on this but this change can fix the issue that query_engine is not constructed.